src/modules/discovery/hosts.py

```diff
@@ -87,61 +105,39 @@ def __init__(self, event):
         self.event = event
 
     def execute(self):
-        # Scan any hosts that the user specified
+        # If user has specified specific remotes, scanning only them
         if config.remote or config.cidr:
             self.publish_event(HostScanEvent())
         else:
-            # Discover cluster subnets, we'll scan all these hosts 
-            if self.is_azure_pod():
-                subnets, cloud = self.azure_metadata_discovery()
-            else:
-                subnets, cloud = self.traceroute_discovery()
+            # figuring out the cloud from the external ip, default to CloudTypes.NO_CLOUD
+            external_ip = HostDiscoveryUtils.get_external_ip()
+            cloud = HostDiscoveryUtils.get_cloud(external_ip)
+
+            # specific cloud discoveries should subscribe to RunningPodOnCloud
+            if cloud != CloudTypes.NO_CLOUD:
+                self.publish_event(RunningPodOnCloud(cloud=cloud))
 
+            # normal pod discovery
+            scan_subnets = self.pod_subnet_discovery()
```

Comment:
Looks like this list will always gonna have a single item.
You can throw the loop over that list.